### PR TITLE
Fix some accessibility issues in the header

### PR DESCRIPTION
**Pull request by alexstine**

### CODE REVIEW

File: components/DarkModeToggle.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const DarkModeToggle = () => {
           id="dark-mode-toggle"
           className="sr-only"
           checked={!isDarkMode}
-          aria-label={"Dark Mode Toggle"}
+          aria-label={"Dark Mode"}
```
Comment (alexstine) [inline components/DarkModeToggle.tsx:37]: Tried to reduce verbosity. Checkbox is implied toggle.

File: components/navbar.tsx [hunk 2]
```diff
@@ -9,8 +9,15 @@ import * as gtag from "../lib/gtag";
 
 export default function NavBar() {
   let [mobileSearchBoxOpen, setMobileSearchBoxOpen] = useState(false);
+  const mobileSearchButton = useRef<HTMLButtonElement>(null);
```
Comment (alexstine) [inline components/navbar.tsx:12]: Next bit of code helps out with focus after mobile search is closed.

File: components/navbar.tsx [hunk 3]
```diff
@@ -61,6 +69,8 @@ export default function NavBar() {
                 onClick={() => {
                   setMobileSearchBoxOpen(true);
                 }}
+                aria-label="Search"
```
Comment (alexstine) [inline components/navbar.tsx:72]: Buttons without labels are no good.

File: components/search/MobileSearchBox.tsx [hunk 4]
```diff
@@ -32,7 +32,6 @@ function SearchBox({
       <input
         id={"algolia_search"}
         type="search"
-        aria-label="Search for commands"
```
Comment (alexstine) [inline components/search/MobileSearchBox.tsx:35]: The HTML label is fine. No need for aria-label to explain the same text.

File: components/search/MobileSearchBox.tsx [hunk 5]
```diff
@@ -44,6 +43,7 @@ function SearchBox({
       <button
         className="h-9 flex items-center text-icon-light dark:text-icon-dark pr-3 pl-1 bg-inherit"
         onClick={onCloseCallback}
+        aria-label="Close Search"
```
Comment (alexstine) [inline components/search/MobileSearchBox.tsx:46]: Need a label for the close button.

**Review by alexstine — COMMENTED** [review]
Hopefully these are minor enough changes to not cause visual regressions. Tested locally, works great.

**Review by ianhodge — APPROVED** [review]
Thank you so much for making these changes :) Learned a bit myself in reviewing this on how to make better a11y labels.

I approved the vercel deployment, but it still seems to be failing. @michlimlim do you have more context here?

**Comment by michlimlim** [comment]
Fixed

**Comment by alexstine** [comment]
@ianhodge Anything left to do here or can you merge this one in? Thanks.

**Comment by ianhodge** [comment]
All good to me! Merging :)

**Pull request merged**